crates/ruff_cli/src/commands/format.rs

```diff
@@ -567,3 +572,69 @@ impl Display for FormatCommandError {
         }
     }
 }
+
+pub(super) fn warn_incompatible_formatter_settings(
+    pyproject_config: &PyprojectConfig,
+    resolver: Option<&Resolver>,
+) {
+    for setting in std::iter::once(&pyproject_config.settings)
+        .chain(resolver.iter().flat_map(|resolver| resolver.settings()))
+    {
+        let mut incompatible_rules = Vec::new();
+
+        for incompatible_rule in RuleTable::from_iter([
+            Rule::LineTooLong,
+            Rule::TabIndentation,
+            Rule::IndentationWithInvalidMultiple,
+            Rule::IndentationWithInvalidMultipleComment,
+            Rule::OverIndented,
+            Rule::IndentWithSpaces,
+            Rule::SingleLineImplicitStringConcatenation,
+            Rule::MissingTrailingComma,
+            Rule::ProhibitedTrailingComma,
+            Rule::BadQuotesInlineString,
+            Rule::BadQuotesMultilineString,
+            Rule::BadQuotesDocstring,
+            Rule::AvoidableEscapedQuote,
+        ])
+        .iter_enabled()
+        {
+            if setting.linter.rules.enabled(incompatible_rule) {
+                incompatible_rules.push(format!("'{}'", incompatible_rule.noqa_code()));
+            }
+        }
+
+        if !incompatible_rules.is_empty() {
+            incompatible_rules.sort();
+            warn!("The following rules can cause conflicts when used with the formatter: {}. We recommend disabling these rules when using Ruff's formatter to avoid unexpected behavior. To disable the rules, change your configuration and either remove the rules from the `select` or `extend-select` options or add the rules to the `ignore` option.", incompatible_rules.join(", "))
```

Comment:
Small nudges on the comment:

```
The following rules may cause conflicts when used with the formatter: {}.
To avoid unexpected behavior, we recommend disabling these rules, either
by removing them from your `select` or `extend-select`, or adding them to
your `ignore`.
```

Reply:
"your ignore" sounds a bit strange. I used your suggestion but changed it to "from the `select` or `extend-select` configuration.